src/server/scripts/Kalimdor/CavernsOfTime/BattleForMountHyjal/boss_archimonde.cpp

```diff
@@ -514,13 +561,14 @@ class boss_archimonde : public CreatureScript
             if (!target)
                 return;
 
-            me->SummonCreature(NPC_DOOMFIRE_SPIRIT,
+            Creature* DoomfireSpirit = reinterpret_cast<Creature*>(me->SummonCreature(NPC_DOOMFIRE_SPIRIT,
                 target->GetPositionX()+15.0f, target->GetPositionY()+15.0f, target->GetPositionZ(), 0,
-                TEMPSUMMON_TIMED_DESPAWN, 27s);
-
-            me->SummonCreature(NPC_DOOMFIRE,
+                TEMPSUMMON_TIMED_DESPAWN, 27s));
+            DoomfireSpirit->SetVisible(false);
+            Creature* Doomfire = reinterpret_cast<Creature*>(me->SummonCreature(NPC_DOOMFIRE,
                 target->GetPositionX()-15.0f, target->GetPositionY()-15.0f, target->GetPositionZ(), 0,
-                TEMPSUMMON_TIMED_DESPAWN, 27s);
+                TEMPSUMMON_TIMED_DESPAWN, 27s));
+            Doomfire->SetVisible(false);
```

Comment:
flags_extra 128 in creature_template is more preferable & legit here, but it can break something, should be tested

Reply:
I test flags_extra 128,seems to work correctly.